### ISSUE TITLE
Move transaction types (eg. `FeeMarketTransaction`, `LegacyTransaction`) into their own file

### ISSUE THREAD

**Issue by SamWilsn** [ethereum/execution-specs#876]
There are enough different types in the more recent forks to justify moving all the transaction classes and helper functions into their own file (maybe `src/ethereum/*/transactions.py`?). `fork_types.py` is just a catch-all, and will continue to grow if we aren't careful.

**Comment by Redidacove** [comment]
Can you specify where exactly you need changes to be done in the repo location.Would try to do it .

**Comment by SamWilsn** [comment]
Sure! So as an example:

https://github.com/ethereum/execution-specs/blob/4d953035fb0cceda7cf21d71b2ab7a9a6f4632f0/src/ethereum/london/fork_types.py#L47-L137

These types would be moved to something like `src/ethereum/london/transactions.py`.

Note that any of these changes need to be repeated for all the forks (eg. `src/ethereum/frontier`, `src/ethereum/homestead`, etc.)

**Comment by Redidacove** [comment]
Ohk sure so I need to make files like transaction.py ,account.py, header.py,log.py and move the related to code to respective files and same thing for other forks as well like frontier ,homestead etc .
Plz correct me if I am wrong.

**Comment by Redidacove** [comment]
Plz reply so that I could work on this @SamWilsn

**Comment by Redidacove** [comment]
Even I did the setup just needed more clarity on this @SamWilsn could you assign this to me this issue plz

**Comment by gurukamath** [comment]
> Ohk sure so I need to make files like transaction.py ,account.py, header.py,log.py and move the related to code to respective files and same thing for other forks as well like frontier ,homestead etc .

For now atleast, I would just re-factor the different transaction types into a separate module since that is the part of the `fork_types.py` that is most likely to grow. The other classes can remain under the umbrella module `fork_types.py` for now.

And these changes will have to be ported to all the forks starting from frontier. Here is what I would suggest

- You could open 1 PR pointed at `master` to make these changes for forks from `Frontier` to `Shanghai`
- Another PR pointed at `forks/cancun` to make these changes to `Cancun`

Hope that answered your queries.

**Comment by Redidacove** [comment]
Thanks for the clarification I will do the above mentioned things

**Comment by Redidacove** [comment]
@gurukamath just a doubt I can't see any folder inside src/ethereum could you plz specify which one is referring to Cancun

**Comment by gurukamath** [comment]
> @gurukamath just a doubt I can't see any folder inside src/ethereum named as Cancun could you plz tell me which one is referring to Cancun

`cancun` is part of the `forks/cancun` branch and will be merged into master only after it goes on mainnet.

**Comment by Redidacove** [comment]
I did it and created a PR plz review it @gurukamath

**Comment by Redidacove** [comment]
#894

**Comment by Redidacove** [comment]
#894 This is my PR for the fix that will close this

**Comment by Redidacove** [comment]
> > Ohk sure so I need to make files like transaction.py ,account.py, header.py,log.py and move the related to code to respective files and same thing for other forks as well like frontier ,homestead etc .
> 
> For now atleast, I would just re-factor the different transaction types (along with related functions) into a separate module since that is the part of the `fork_types.py` that is most likely to grow. The other classes can remain under the umbrella module `fork_types.py` for now.
> 
> And these changes will have to be ported to all the forks starting from frontier. Here is what I would suggest
> 
>     * You could open 1 PR pointed at `master` to make these changes for forks from `Frontier` to `Shanghai`
> 
>     * Another PR pointed at `forks/cancun` to make these changes to `Cancun`
> 
> 
> Hope that answered your queries.

I am not able to figure out how can i fork fork/cancun branch code and work with that and do a pr pointing to fork/cancun I know its a beginner question but I am not able to figure this out

**Comment by SamWilsn** [comment]
Yep, I'm working on it!
